extensions/frontend-tiles/src/BatchedTile.ts

```diff
@@ -0,0 +1,184 @@
+    super(params, tree);
+    if (params.childrenProps?.length)
+      this._childrenProps = params.childrenProps;
+
+    if (!this.contentId)
+      this.setIsReady();
+  }
+
+  private get _batchedChildren(): BatchedTile[] | undefined {
+    return this.children as BatchedTile[] | undefined;
+  }
+
+  public override computeLoadPriority(viewports: Iterable<Viewport>, _users: Iterable<TileUser>): number {
+    // Prioritize tiles closer to camera and center of attention (zoom point or screen center).
+    return RealityTileLoader.computeTileLocationPriority(this, viewports, this.tree.iModelTransform);
+  }
+
+  public selectTiles(selected: BatchedTile[], args: TileDrawArgs): void {
+    const vis = this.computeVisibility(args);
+    if (TileVisibility.OutsideFrustum === vis)
+      return;
+
+    // ###TODO proper tile refinement. Currently we simply load each level of the tree in succession until we find a tile that
+    // meets screen space error. Moreover, while we wait for children to load we stop displaying the parent.
+    // Prefer to skip some levels where appropriate.
+    // More importantly, fix tile tree structure so that children can be substituted for (portions of) parents more quickly, especially near the camera
+    // (need non-overlapping child volumes).
+    if (!this.isReady) {
+      args.insertMissing(this);
+      return;
+    }
+
+    if (TileVisibility.Visible === vis && this.hasGraphics) {
+      args.markReady(this);
+      selected.push(this);
+      return;
+    }
+
+    args.markUsed(this);
+    if (this.isReady) {
+      const childrenStatus = this.loadChildren();
+      if (TileTreeLoadStatus.Loading === childrenStatus)
+        args.markChildrenLoading();
+
+      const children = this._batchedChildren;
+      if (children)
+          for (const child of children)
+            child.selectTiles(selected, args);
+    }
+  }
+
+  protected override _loadChildren(resolve: (children: Tile[] | undefined) => void, reject: (error: Error) => void): void {
+    let children: BatchedTile[] | undefined;
+    if (this._childrenProps) {
+      try {
+        for (const childProps of this._childrenProps) {
+          const params = this.batchedTree.reader.readTileParams(childProps, this);
+          const child = new BatchedTile(params, this.batchedTree);
+          children = children ?? [];
+          children.push(child);
+        }
+      } catch (err) {
+        Logger.logException(loggerCategory, err);
+        children = undefined;
+        if (err instanceof Error)
+          reject(err);
+      }
+    }
+
+    resolve(children);
+  }
+
+  public override get channel(): TileRequestChannel {
+    return IModelApp.tileAdmin.channels.getForHttp("itwinjs-batched-models");
+  }
+
+  public override async requestContent(_isCanceled: () => boolean): Promise<TileRequest.Response> {
+    const url = new URL(this.contentId, this.batchedTree.reader.baseUrl);
```

Comment:
FYI - This may cause problems if we try to use as-is.  Locally, I've been explicitly appending a ".gz" suffix to the contentId, under the assumption that we'd either update the mesh export service to not append that suffix when uploading or the manifests would be updated to include it in the contentIds...

Reply:
If export service is adding ".gz" to the content URLs then the content URLs embedded in the tileset JSON need to include that suffix too, per the spec.
Preferably, the ".gz" extension will not be present.
Did you confer with @danieliborra?

Reply:
We talked a bit about adding the suffix to the tileset JSON, but my preference would also be to just drop the .gz extension before upload.  We just need to make sure we can do that without breaking existing 3DFT consumers.

Reply:
Let's talk about it today.
I'd also prefer not using the .gz suffix. To avoid breaking changes, I'd add a property to the mesh export service request with the version of the 3DFT format, and update the UE plugin. That would avoid future problems too.
@diegopinate does it make sense?
We can tell the users to update the plugin asap and deprecate the ".gz" version in a couple of months.

Reply:
Yes, we can add a version field and change the blob client to upload non .gz. Do we want to keep compression? We use zlib's deflate to make into a gz archive.

See [this](https://dev.azure.com/bentleycs/iModelTechnologies/_git/graphics-exporter-services?path=packages/export-3dft-agent/src/storage/AzureBlobClient.ts&version=GC5041cd4e0e69a52d072d4017afe77185efc0156f&line=56&lineEnd=92&lineStartColumn=1&lineEndColumn=4&_a=contents)

Reply:
Yes, we should definitely still compress - as long as we're setting the correct `blobContentEncoding` on upload (which we are) then adding the extension is unnecessary.